src/Type/Comment/CommentType.php

```diff
@@ -81,7 +81,23 @@ private static function fields() {
 						'type' => Types::user(),
 						'description' => __( 'The post field for comments matches the post id the comment is assigned to. This field is equivalent to WP_Comment->comment_post_ID and the value matching the `comment_post_ID` column in SQL.', 'wp-graphql' ),
 						'resolve' => function( \WP_Comment $comment, $args, AppContext $context, ResolveInfo $info ) {
-							$author = new \WP_User( $comment->user_id );
+							if ( ! empty( $comment->user_id ) ) {
+								$author = new \WP_User( $comment->user_id );
+							} else {
+								$author = new \WP_User();
+
+								if ( ! empty( $comment->comment_author ) ) {
+									$author->__set( 'display_name', $comment->comment_author );
+								}
+								if ( ! empty( $comment->comment_author_email ) ) {
+									$author->__set( 'user_email', $comment->comment_author_email );
+									$comment_author_id = [ 'type' => 'comment_author', 'email' => $comment->comment_author_email, 'name' => $comment->comment_author ];
```

Comment:
So it looks like you can't set the ID like this. . .but this works: 

```
$author->ID = wp_json_encode([
    'type' => 'comment_author',
    'email' => ! empty( $comment->comment_author_email ) ? $comment->comment_author_email : null,
    'name' => ! empty( $comment->comment_author ) ? $comment->comment_author : null,
]);
```

Then we also need to include a test for the `id` field to make sure it's outputting what we expect, which should look something like: 

`dXNlcjp7InR5cGUiOiJjb21tZW50X2F1dGhvciIsImVtYWlsIjoiamFzb25iYWhsQG1hYy5jb20iLCJuYW1lIjoiSmFzb24gQmFobCJ9`

That's a base64encoded string that contains the following data when decoded:

`user:{"type":"comment_author","email":"jasonbahl@mac.com","name":"Jason Bahl"}`

That ID now contains enough data for the server to be able to identify how to re-fetch the data if needed, and it's unique enough for client side caches to invalidate when that comment author sees a mutation in the client.